librz/asm/p/asm_ppc_gnu.c

```diff
@@ -50,7 +51,14 @@ static int disassemble(RzAsm *a, RzAsmOp *op, const ut8 *buf, int len) {
 
 	/* prepare disassembler */
 	memset(&disasm_obj, '\0', sizeof(struct disassemble_info));
-	disasm_obj.disassembler_options = (a->bits == 64) ? "64" : "";
+	*options = 0;
+	if (!RZ_STR_ISEMPTY(a->cpu)) {
+		snprintf(options, sizeof(options), "%s,%s",
+			(a->bits == 64) ? "64" : "", a->cpu);
+	} else if (a->bits == 64) {
+		rz_str_ncpy(options, "64", sizeof(options));
+	}
+	disasm_obj.disassembler_options = rz_str_newf("%s", options);
```

Comment:
can't you just do `disasm_obj.disassembler_options = rz_str_newf("%s,%s", (a->bits == 64) ? "64" : "", a->cpu);` in that if?

Reply:
I think we can. We'll just eliminate `options[64]` and directly assign `rz_str_newf` on both of the conditions, then. I'll try that.

Reply:
It is fine, just please use `RZ_STR_NOTEMPTY(a->cpu)` instead of `!RZ_STR_ISEMPTY()`